WooCommerce/Classes/Authentication/Store Creation/Profiler/Challenges/StoreCreationChallengesQuestionViewModel.swift

```diff
@@ -0,0 +1,74 @@
+import Combine
+import Foundation
+
+/// Necessary data from the answer of the store creation challenges question.
+struct StoreCreationChallengesAnswer: Equatable {
+    /// Display name of the selected challenge.
+    let name: String
+    /// Raw value of the challenge to be sent to the backend.
+    let value: String
+}
+
+/// View model for `StoreCreationChallengesQuestionView`, an optional profiler question about challenges in the store creation flow.
+@MainActor
+final class StoreCreationChallengesQuestionViewModel: StoreCreationProfilerQuestionViewModel, ObservableObject {
+    typealias Answer = StoreCreationChallengesAnswer
+
+    let topHeader = Localization.topHeader
+
+    let title = Localization.title
+
+    let subtitle = Localization.subtitle
+
+    @Published private(set) var selectedChallenges: [Challenge] = []
+
+    private let onContinue: ([Answer]) -> Void
+    private let onSkip: () -> Void
+
+    init(onContinue: @escaping ([Answer]) -> Void,
+         onSkip: @escaping () -> Void) {
+        self.onContinue = onContinue
+        self.onSkip = onSkip
+    }
+}
+
+extension StoreCreationChallengesQuestionViewModel: OptionalStoreCreationProfilerQuestionViewModel {
+    func continueButtonTapped() async {
```

Comment:
Potential improvement: I don't think we are sending answers immediately, so it's probably worth it to check again if we can remove the `async` requirement for this method.

Reply:
Noted this down as a subtask https://github.com/woocommerce/woocommerce-ios/issues/10374